packages/components/addon/components/hds/dropdown/index.js

```diff
@@ -0,0 +1,60 @@
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+import { htmlSafe } from '@ember/template';
+
+export const DEFAULT_POSITION = 'right';
+export const POSITIONS = ['right', 'left'];
+
+export default class HdsDropdownIndexComponent extends Component {
+  /**
+   * @param listPosition
+   * @type {string}
+   * @default primary
+   * @description Determines the position of the "list"
+   */
+  get listPosition() {
+    let { listPosition = DEFAULT_POSITION } = this.args;
+
+    assert(
+      `@listPosition for "Hds::Dropdown::Index" must be one of the following: ${POSITIONS.join(
+        ', '
+      )}; received: ${listPosition}`,
+      POSITIONS.includes(listPosition)
+    );
+
+    return listPosition;
+  }
+
+  /**
+   * Get a style attribute to apply to the "list" based on the @width argument
+   * @return {string} The style attribute to apply to the list
+   */
+  get listStyle() {
+    return this.args.width ? htmlSafe(`width: ${this.args.width}`) : undefined;
```

Comment:
Counter-intuitively, Ember’s [`htmlSafe`](https://api.emberjs.com/ember/4.2/functions/@ember%2Ftemplate/htmlSafe) function does not perform any sanitization, but rather tells Ember “this string is already safe, no need to sanitize it”.

Reply:
I agree with @jgwhite.  `htmlSafe` should be avoided, especially with input data.  I'm curious why this is even necessary though?  Would any part of the concatenated string be sanitized by Ember?

Reply:
> I was mostly wondering what kind of input data you were expecting that would even be subject to sanitization.
No user-generated data, just consumers (you) values.

Reply:
OK, adopted `ember-style-modifier` instead of doing inline styling ourselves.